etna/libs/chronos/chronos_bolt.py

```diff
@@ -0,0 +1,752 @@
+            built-in prediction length from the model. False by
+            default. When true, fail loudly if longer predictions
+            are requested, otherwise longer predictions are allowed.
+
+        Returns
+        -------
+        torch.Tensor
+            Forecasts of shape (batch_size, num_quantiles, prediction_length)
+            where num_quantiles is the number of quantiles the model has been
+            trained to output. For official Chronos-Bolt models, the value of
+            num_quantiles is 9 for [0.1, 0.2, ..., 0.9]-quantiles.
+
+        Raises
+        ------
+        ValueError
+            When limit_prediction_length is True and the prediction_length is
+            greater than model's training prediction_length.
+        """
+        context_tensor = self._prepare_and_validate_context(context=context)
+
+        model_context_length = self.model.config.chronos_config["context_length"]
+        model_prediction_length = self.model.config.chronos_config["prediction_length"]
+        if prediction_length is None:
+            prediction_length = model_prediction_length
+
+        if prediction_length > model_prediction_length:
+            msg = (
+                f"We recommend keeping prediction length <= {model_prediction_length}. "
+                "The quality of longer predictions may degrade since the model is not optimized for it. "
+            )
+            if limit_prediction_length:
+                msg += "You can turn off this check by setting `limit_prediction_length=False`."
+                raise ValueError(msg)
+            warnings.warn(msg)
+
+        predictions = []
+        remaining = prediction_length
+
+        # We truncate the context here because otherwise batches with very long
+        # context could take up large amounts of GPU memory unnecessarily.
+        if context_tensor.shape[-1] > model_context_length:
+            context_tensor = context_tensor[..., -model_context_length:]
+
+        # TODO: We unroll the forecast of Chronos Bolt greedily with the full forecast
+        # horizon that the model was trained with (i.e., 64). This results in variance collapsing
+        # every 64 steps.
+        while remaining > 0:
+            prediction = []
+            for i in range(0, context_tensor.shape[0], batch_size):
+                batch_context_tensor = context_tensor[i : i + batch_size, :].to(
+                    device=self.model.device,
+                    dtype=torch.float32,  # scaling should be done in 32-bit precision
+                )
+                with torch.no_grad():
+                    batch_prediction = self.model(  # added batch iteration
+                        context=batch_context_tensor,
+                    ).quantile_preds.to(batch_context_tensor)
+                prediction.append(batch_prediction)
+
+            prediction = torch.cat(prediction, dim=0)
+            predictions.append(prediction)
```

Comment:
Maybe we can do some fixes here of authors' code?
As I understand, if code is computed on gpu, the full tensor will be on gpu and only in the end it will be transferred to cpu.
The suggested fix is to transfer output batch to cpu.